plugin/pkg/admission/persistentvolume/label/admission.go

```diff
@@ -40,8 +41,9 @@ var _ = admission.Interface(&persistentVolumeLabel{})
 type persistentVolumeLabel struct {
 	*admission.Handler
 
-	mutex      sync.Mutex
-	ebsVolumes aws.Volumes
+	mutex            sync.Mutex
+	ebsVolumes       aws.Volumes
+	gceCloudProvider *gce.GCECloud
```

Comment:
I don't see how - suggestions welcome.

I wanted to implement a Volumes interface, and you can see from the AWS implementation that I did.  It was originally intended to be cross-cloud provider, but there were a number of objections raised so I moved Volumes into the aws package.  The primary objection was (I believe) that a cloud could implement different types of volumes (e.g. even AWS has NFS & EBS now).

And volume.Spec isn't provider independent either.

One option would be to define aws.Volumes in cloudprovider, but have it take a PersistentVolumeSource.  But that isn't quite right either, and would be a fairly large refactor.

In short, I'd love to, but I don't see how to do it in an acceptable way.